src/net_processing.cpp

```diff
@@ -238,7 +240,7 @@ struct CNodeState {
     //! Time of last new block announcement
     int64_t m_last_block_announcement;
 
-    CNodeState(CAddress addrIn, std::string addrNameIn) : address(addrIn), name(addrNameIn) {
+    CNodeState(CNode* connectionIn) : connection(connectionIn), address(connection->addr), name(connection->GetAddrName()) {
```

Comment:
I agree with passing an opaque handle in directly here, but making it easier for net_processing to use CNode's internals is the _opposite_ of the direction we should be headed in :(

Reply:
Sure, but incremental steps - its clearly the right direction, even if CNode will have more fields made private over the coming months...its not like we have to merge things which reference more CNode fields aside from clear const descriptors that just describe the connection itself.